lib/gui/components/update-notifier.js

```diff
@@ -138,17 +166,48 @@ module.exports = function($http, $q, ModalService, UPDATE_NOTIFIER_SLEEP_DAYS, M
    */
   this.notify = () => {
     return this.getLatestVersion().then((version) => {
-      return ModalService.open({
-        template: './components/update-notifier/templates/update-notifier-modal.tpl.html',
-        controller: 'UpdateNotifierController as modal',
-        size: 'update-notifier',
-        resolve: {
-          options: _.constant({
-            version
-          })
+      settings.set('lastUpdateNotify', Date.now());
+      settings.set('sleepUpdateCheck', false);
```

Comment:
I don't think I get it. The `lastUpdateNotify` option represents the date where the update notification last appeared, so it makes sense that it has to be set to the current time right before showing the dialog, and it will only get set again once we determine that the dialog should be shown again (e.g: after 7 days if the sleep update check was set).

> anytime the dialog appears it'd pause it for seven days without the help of sleepUpdateCheck

Can you capture this as a unit test so I can better understand the issue? I might be missing something, but we only call `.notify()` if `.shouldCheckForUpdates()` returns `true`.

If we show the dialog once, then `lastUpdateNotify` has the current date. Lets say the user didn't set `sleepUpdateCheck`. The user will then open Etcher again the day afterwards. At this point, `lastUpdateNotify` contains the day before time, and `sleepUpdateCheck` is `false`. The `(!settings.get('sleepUpdateCheck') || !lastUpdateNotify)` is `true`, which makes `.shouldCheckForUpdates()` return `true`, and then the dialog is shown (note that the `lastUpdateNotify` is not even considered, given that `!settings.get('sleepUpdateCheck')` is already `true`).

Now lets consider the same scenario if the user did set `sleepUpdateCheck`. The first conditonal, `(!settings.get('sleepUpdateCheck') || !lastUpdateNotify)` is false, because `sleepUpdateCheck` is set and `lastUpdateNotify` has a value, which makes `!lastUpdateNotify` evaluate to `false`.

So the next condition applies: 

```js
if (lastUpdateNotify - Date.now() > units.daysToMilliseconds(this.UPDATE_NOTIFIER_SLEEP_DAYS)) {
  settings.set('sleepUpdateCheck', false);
  return true;
}
```

If the date set on `lastUpdateNotify` is greater than 7 days, then the dialog is shown, and only then the `lastUpdateNotify` variable gets set again. Otherwise, the function returns `false`, and the update notifier is not shown.

Am I missing something?